osu.Game.Rulesets.Osu/Utils/OsuHitObjectGenerationUtils_Reposition.cs

```diff
@@ -287,6 +316,39 @@ private static Vector2 clampToPlayfieldWithPadding(Vector2 position, float paddi
             );
         }
 
+        /// <summary>
+        /// Estimate the centre of mass of a slider relative to its start position.
+        /// </summary>
+        /// <param name="slider">The slider to process.</param>
+        /// <returns>The centre of mass of the slider.</returns>
+        private static Vector2 calculateCentreOfMass(Slider slider)
+        {
+            if (slider.Distance < 1) return Vector2.Zero;
+
+            int count = 0;
+            Vector2 sum = Vector2.Zero;
+            double pathDistance = slider.Distance;
+
+            for (double i = 0; i < pathDistance; i++)
+            {
+                sum += slider.Path.PositionAt(i / pathDistance);
+                count++;
+            }
```

Comment:
Not really sure about this method of estimation. It's going to end up checking more points for longer sliders, and there are some maps with very long sliders out there. I'd advocate for a bigger step than that at least, or for always sampling a fixed number of evenly-spaced points. Yes the result will be worse, but there's a big potential for things blowing up here, so worth preemptively addressing, I think?

Reply:
I've increased the step to 50. This is now sampling only a few points for most sliders, which is already enough.